File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        INT_MIN, INT_MAX = -2**31, 2**31 - 1
        sign = 1
        if x<0:
            x=x*-1
            sign=-1
        final_num=0
        while x!=0:
            digit=x%10
            final_num=final_num*10+digit
            x=x//10
        final_num*=sign
        if final_num < INT_MIN or final_num > INT_MAX:
            return 0
        return final_num

File: 7-reverse-integer/reverse-integer.cpp
#include <iostream>
#include <climits>  // For INT_MIN and INT_MAX

class Solution {
public:
    int reverse(int x) {
        int sign = 1;
        if (x < 0) {
            sign = -1;
        }

        int final = 0;
        int digit;
        while (x != 0) {
            digit = x % 10;
            // Check if reversing will cause overflow
            if (final > INT_MAX / 10 || (final == INT_MAX / 10 && digit > 7)) {
                return 0;  // Return 0 if overflow would occur
            }
            if (final < INT_MIN / 10 || (final == INT_MIN / 10 && digit < -8)) {
                return 0;  // Return 0 if overflow would occur
            }
            final = final * 10 + digit;
            x = x / 10;
        }
        int num=final*sign;
        return num*sign;
    }
};
